lib/chai/utils/index.js

```diff
@@ -94,3 +94,7 @@ export {isNaN} from './isNaN.js';
 
 // getOperator method
 export {getOperator} from './getOperator.js';
+
+export function isRegExp(obj) {
+  return Object.prototype.toString.call(obj) === '[object RegExp]';
+}
```

Comment:
Why is this needed?

Reply:
A RegExp from a virtual context isn't the same RegExp as in the owning context, so instanceof won't work

If you pass chai into the virtual context, it'll be comparing against it's own RegExp rather than that of the virtualised code